ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt

```diff
@@ -616,22 +618,50 @@ private fun MainStackComponent(
     }
 
     if (nestedBadge == null && overlay == null) {
-        stack(
-            outerShapeModifier
-                .then(borderModifier)
-                .then(innerShapeModifier)
-                .conditional(stackState.applyBottomWindowInsets) {
-                    windowInsetsPadding(systemBarInsets.only(WindowInsetsSides.Bottom))
-                },
-        )
+        if (backgroundStyle is BackgroundStyle.Video) {
+            // Video backgrounds require a Box wrapper with explicit sizing
+            Box(
```

Comment:
Hmm I'm wondering, could this work by passing these modifiers to the box inside the `ViewWithVideoBackground` composable, instead of wrapping the `WithOptionalVideoBackground` inside a `Box`. Feels like a level on indirection that might not be needed?